### PR TITLE
Add new scoreboard entries over entries with same time. (Fixes #12)

### LINKED ISSUE #12: Visual Bug in the game history after finishing a game
After successfully solving a sudoko in the SAME amount of time as a previously solved sudoku, the earliest game with this time gets highlighted instead of the current one. See screenshot

Screenshot_20230113-172323__01

### PR DESCRIPTION
**Pull request by AstralShadow**
Also applying the fix from #11 to the generated scoreboard after a game is won.
(Turns out there are 2 places where the leaderboard is being edited: once when saving, and once before displaying after a winning game.)

If there is a need i can post these as 2 separate pull requests.

### CODE REVIEW

File: lib/game.dart [hunk 1]
```diff
@@ -813,11 +813,11 @@ class _SudokuGameState extends State<SudokuGame> with TickerProviderStateMixin {
 
           // also add it to our local list
           DateTime now = DateTime.now();
-          scores.add(Score(_stopwatch.elapsed + _stopwatchOffset, "${now.day} ${DateFormat.yMMM().format(now)}"));
+          scores.add(0, Score(_stopwatch.elapsed + _stopwatchOffset, "${now.day} ${DateFormat.yMMM().format(now)}"));
```
Comment (TheSunCat) [inline lib/game.dart:816]: I don't understand what this 0 argument is doing. I've never seen a second argument in List::add, and the [reference](https://api.dart.dev/be/180791/dart-core/List/add.html) doesn't mention it either.

Comment (AstralShadow) [reply lib/game.dart:816]: Sorry, i've been mistaken with java [reference](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/List.html#add(int,E))

What i meant to do was inserting the element at first position.
This functionality is possible with [insert](https://api.dart.dev/be/180791/dart-core/List/insert.html) according to dart reference.

I'll modify the request accordingly.

**Comment by TheSunCat** [comment]
Ahh I see, this will insert new entries earlier so as to highlight the latest one when two scores match. Have you checked that this works? I'm concerned that scores.sort call right after might mess with the order (though I haven't tested it myself).

**Comment by AstralShadow** [comment]
Sadly, the dart List.sort says:
> The sort function is not guaranteed to be stable, so distinct objects that compare as equal may occur in any order in the result.

Can we assume that highscores are stored after being sorted in current version?
If we do assume this, we can just insert the new highscore at it's correct position with the insert call, and we would not need to sort at all.

**Comment by AstralShadow** [comment]
I implemented what i mean, but i dont have flutter configured, so could you do a compile-test and tell me if i messed up something?

**Comment by TheSunCat** [comment]
Of course, though I'm busy with finals for two or so weeks at the moment. If I find a time before then I will be sure to test it!

**Comment by AstralShadow** [comment]
> Of course, though I'm busy with finals for two or so weeks at the moment. If I find a time before then I will be sure to test it!

Good luck on your finals!

**Review by TheSunCat — CHANGES_REQUESTED** [review]
uhh I meant to request these changes, but it seems I pushed them. Gotta learn how this stuff works heh.

Anyhow, I've tested it and it appears to be working, so should be good to merge now!

**Review by TheSunCat — APPROVED** [review]

**Comment by AstralShadow** [comment]
How do you know if `a` or `b` is the new time?
Sort() is not sure to be stable, even if it seems to be when testing, so we can't be sure which of the two elements is the one we add.

My initial idea was that if we always keep the scoreboard sorted and we insert new scores at their proper position, it will stay sorted and not need any extra sorting.

**Comment by TheSunCat** [comment]
Hmm, I think you're right. I had assumed the comparisons would happen in order of the list, but I can't find anywhere this is guaranteed. So even if it works now, it might not in future versions of Flutter. I will change it to your implementation, as it works well in this case. I'll commit the fix directly to master.
Sorry about merging it so fast! To be honest, I have never handled proper pull requests before.

**Comment by AstralShadow** [comment]
Can you do a version bump & release (so F-Droid shows up an update) when you got the time?
Last version is from Oct 27, 2022.

**Comment by TheSunCat** [comment]
Thanks for the reminder! I meant to wait to ship the tutorial "onboarding" screen with the next release, but that took much longer than expected and I've not had time to work on it due to university, so I will just release the bugfixes that we have now.

**Comment by AstralShadow** [comment]
Thanks for the release! 
For some reason it doesn't show up as latest release. (a cause might be it being a draft, beta, or the project using manual latest release selection)
As far as i know F-Droid picks the release marked as latest, so could you try to mark it as such?

**Comment by TheSunCat** [comment]
Done! Marked it as the latest release, hopefully that does it.

**Pull request merged**